src/Model/Aggregate.php

```diff
@@ -0,0 +1,316 @@
+                $query = $this->baseModel->action($mode, [false]);
+
+                $this->initQueryFields($query, array_unique($fields + $this->systemFields));
+                $this->initQueryOrder($query);
+                $this->initQueryGrouping($query);
+                $this->initQueryConditions($query);
+
+                $this->hook(self::HOOK_INIT_SELECT_QUERY, [$query]);
+
+                return $query;
+            case 'count':
+                $query = $this->baseModel->action($mode, $args);
+
+                $query->reset('field')->field($this->expr('1'));
+                $this->initQueryGrouping($query);
+
+                $this->hook(self::HOOK_INIT_SELECT_QUERY, [$query]);
+
+                return $query->dsql()->field('count(*)')->table($this->expr('([]) der', [$query]));
+            case 'field':
+            case 'fx':
+                return parent::action($mode, $args);
+            default:
+                throw (new Exception('Aggregate model does not support this action'))
+                    ->addMoreInfo('mode', $mode);
+        }
+    }
+
+    protected function initQueryFields(Query $query, array $fields = []): void
+    {
+        $this->persistence->initQueryFields($this, $query, $fields);
+    }
+
+    protected function initQueryOrder(Query $query): void
+    {
+        if ($this->order) {
+            foreach ($this->order as $order) {
+                $isDesc = strtolower($order[1]) === 'desc';
+
+                if ($order[0] instanceof Expression) {
+                    $query->order($order[0], $isDesc);
+                } elseif (is_string($order[0])) {
+                    $query->order($this->getField($order[0]), $isDesc);
+                } else {
+                    throw (new Exception('Unsupported order parameter'))
+                        ->addMoreInfo('model', $this)
+                        ->addMoreInfo('field', $order[0]);
+                }
+            }
+        }
+    }
+
+    protected function initQueryGrouping(Query $query): void
+    {
+        // use table alias of base model
+        $this->table_alias = $this->baseModel->table_alias;
+
+        foreach ($this->group as $field) {
+            if ($this->baseModel->hasField($field)) {
+                $expression = $this->baseModel->getField($field);
+            } else {
+                $expression = $this->expr($field);
+            }
+
+            $query->group($expression);
+        }
+    }
+
+    protected function initQueryConditions(Query $query, Model\Scope\AbstractScope $condition = null): void
+    {
+        $condition = $condition ?? $this->scope();
+
+        if (!$condition->isEmpty()) {
+            // peel off the single nested scopes to convert (((field = value))) to field = value
+            $condition = $condition->simplify();
+
+            // simple condition
+            if ($condition instanceof Model\Scope\Condition) {
+                $query->having(...$condition->toQueryArguments());
```

Comment:
`having` is an interesting argument for `Aggregate` class, but we may use HAVING instead or WHERE when grouping is active in general (before we can analyse if HAVING is really needed)

Reply:
Ok for now if we will integrate it for wrapping like now...

Reply:
I guess `having` is always needed in place of `where` in case model is grouped and condition is applied on such aggregate model itself.

Reply:
not if we wrap as subquery...

Reply:
Related with https://github.com/atk4/data/pull/853/files#diff-7e7810cb7d196a13e4c0b10d1c737e5ead3e3169bd82f6b4c1149dcf67d726f0R384 , I belive wrapping is the only solution, `having` requires `group by` across all columns for some DB vendors (even if grouped by some unique column)

Reply:
TODO after https://github.com/atk4/data/pull/946